django-stubs/db/models/constraints.pyi

```diff
@@ -44,11 +44,24 @@ class BaseConstraint:
 
 class CheckConstraint(BaseConstraint):
     check: Q | BaseExpression
+    condition: Q | BaseExpression
+
+    @deprecated("The check keyword argument is deprecated in favor of condition and will be removed in Django 6.0")
+    @overload
```

Comment:
PEP 702 [suggests](https://peps.python.org/pep-0702/#example) using `@overload` and then `@deprecated`

Reply:
Updated